src/assertions.cpp

```
@@ -194,8 +200,8 @@ class AssertExtractor {
     vector<Assert> operator()(const Atomic& ins) const {
         vector<Assert> res;
         res.emplace_back(TypeConstraint{ins.access.basereg, TypeGroup::pointer});
-        res.emplace_back(
-            ValidAccess{ins.access.basereg, ins.access.offset, Imm{static_cast<uint32_t>(ins.access.width)}, false});
+        res.emplace_back(make_valid_access(ins.access.basereg, ins.access.offset,
+                                           Imm{static_cast<uint32_t>(ins.access.width)}, false));
```

cppcheck

[error] Syntax Error

(internalAstError)